### PR TITLE
BUGFIX: SD-922 should re-render geo chart when config or data source changed

### PR DESCRIPTION
**Pull request by lvkhanh**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/3003
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2720

# Related Jira tasks

### CODE REVIEW

**Comment by gdgate** [comment]
Merge Failed.

Zuul merger could not merge this change into the base branch. This is most likely caused by merge conflicts. Please rebase the change and upload the rebased version. In case of further errors, contact the zuul administrator.

**Comment by lvkhanh** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1536.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6963/ : FAILURE in 4m 05s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2878/ : SUCCESS in 4s

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by lvkhanh** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1536.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6966/ : SUCCESS in 10m 52s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2882/ : SUCCESS in 4s

File: src/components/core/base/GeoValidatorHOC.tsx [hunk 1]
```diff
@@ -80,6 +83,25 @@ export function geoValidatorHOC<T>(InnerComponent: React.ComponentClass<T>): Rea
 
             return <ErrorComponent code={errorState} {...errorProps} />;
         }
+
+        private isSameConfig(config: IGeoConfig, nextConfig: IGeoConfig): boolean {
+            const colorMapping = (config.colorMapping || []).map(
+                (currentColor: IColorMapping) => currentColor.color,
```
Comment (naibinh) [inline src/components/core/base/GeoValidatorHOC.tsx:89]: type for predicate :)
`(currentColor: IColorMapping): IColorItem => currentColor.color,`

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2883/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3012/ : SUCCESS in 36m 24s

File: src/components/core/base/GeoValidatorHOC.tsx (hunk 1 above)
Comment (lvkhanh) [reply src/components/core/base/GeoValidatorHOC.tsx]: updated

**Comment by lvkhanh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6968/ : SUCCESS in 6m 24s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2887/ : SUCCESS in 6s

**Comment by naibinh** [comment]
LGTM

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3643/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2888/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3015/ : SUCCESS in 18m 19s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6972/ : SUCCESS in 3m 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2892/ : SUCCESS in 3s

**Comment by ptquang86** [comment]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3648/ : SUCCESS in 20m 37s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6977/ : SUCCESS in 12m 00s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1689/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2453/ : SUCCESS in 29m 44s

**Pull request merged**